Over-cloning in code which results in unintended behaviour

```
class Main {
    @ property
    var a = 0;

    @ constructor
    fun() => { 
        a = a + 1;
        sum_to_a(5);
    }

    @ method
    fun sum_to_a(new_a) => {
    	a = new_a;
    }
}

var main_ins = new Main();
println(main_ins);

main_ins.sum_to_a(4);
println(main_ins);
```
The output will be something like
```shell
{
     a: 5,
     sum_to_a: fun sum_to_a(new_a)
}
Runtime Error: Variable not found to be reassigned
File examples/main.bzs, line 13

 
    a = new_a
    ^^^^^^^^^^
Traceback (most recent call last):
    at Function<sum_to_a>
    at examples/main.bzs
```

Here what happens is that after the class in initialized i.e the constructor is ran it returns a Object of the properties and it's methods but this results in popping of the context, so when the function is accessed after initialization, the method is returned but the context which consisted of the properties and other methods in not present.
[The Class Initialization  related code](https://github.com/BlazifyOrg/blazescript/blob/main/src/core/interpreter/value.rs#L381)

Any help or pull requests are greatly appreciated...

Logic has been fixed in the commit https://github.com/BlazifyOrg/blazescript/commit/012be9436619df98e1f212863c1ec69c61af837b but rust is still screaming with the errors of 
```shell
error[E0507]: cannot move out of a shared reference
   --> src/core/interpreter/interpreter.rs:363:35
    |
363 |                     object = Some(*self.ctx.last().unwrap().symbols.get("soul").unwrap());
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ move occurs because value has type `Symbol`, which does not implement the `Copy` trait

error[E0382]: borrow of moved value: `properties_e`
   --> src/core/interpreter/value.rs:567:17
    |
548 |             let mut properties_e: HashMap<String, Value> = HashMap::new();
    |                 ---------------- move occurs because `properties_e` has type `HashMap<String, Value>`, which does not implement the `Copy` trait
...
556 |                         properties: properties_e,
    |                                     ------------ value moved here
...
567 |                 properties_e.insert(k.to_string(), e_v.val.unwrap());
    |                 ^^^^^^^^^^^^ value borrowed here after move

error[E0507]: cannot move out of a shared reference
   --> src/core/interpreter/value.rs:583:24
    |
583 |             let soul = inter.ctx.last().unwrap().symbols.get("soul").unwrap().value;
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                        |
    |                        move occurs because value has type `Value`, which does not implement the `Copy` trait
    |                        help: consider borrowing here: `&inter.ctx.last().unwrap().symbols.get("soul").unwrap().value`

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0382, E0507.
For more information about an error, try `rustc --explain E0382`.
error: could not compile `blazescript`
```

In the latest commit the Object property modifying has been fixed but rust is back again at screaming
```shell
error[E0308]: mismatched types
   --> src/core/interpreter/interpreter.rs:605:29
    |
605 |                 return Some(sym.unwrap());
    |                             ^^^^^^^^^^^^ expected struct `Symbol`, found `&Symbol`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `blazescript`
```

--- a/src/core/interpreter/interpreter.rs
+++ b/src/core/interpreter/interpreter.rs
@@ -154,7 +154,7 @@ impl Interpreter {
 
                 self.ctx.last_mut().unwrap().symbols.insert(
                     name.value.into_string(),
-                    Symbol::new(val.val.unwrap(), reassignable),
+                    Symbol::new(&val.val.unwrap(), reassignable),
                 );
                 res.success(Value::Null)
             }
@@ -196,7 +196,7 @@ impl Interpreter {
 
                 self.get_and_set_ctx(
                     name.value.into_string(),
-                    Symbol::new(val.val.unwrap(), true),
+                    Symbol::new(&val.val.unwrap(), true),
                 );
 
                 res.success(Value::Null)
@@ -294,7 +294,7 @@ impl Interpreter {
                     self.ctx.last_mut().unwrap().symbols.insert(
                         var_name_token.clone().value.into_string(),
                         Symbol::new(
-                            Value::Int {
+                            &Value::Int {
                                 val: i,
                                 pos_start,
                                 pos_end,
@@ -360,7 +360,7 @@ impl Interpreter {
                         );
                     }
 
-                    object = Some(*self.ctx.last().unwrap().symbols.get("soul").unwrap());
+                    object = Some((*self.ctx.last().unwrap().symbols.get("soul").unwrap()).clone());
                 }
 
                 let val = Value::Function {
@@ -379,7 +379,7 @@ impl Interpreter {
                 {
                     self.ctx.last_mut().unwrap().symbols.insert(
                         func_name.clone().value.into_string(),
-                        Symbol::new(val, true),
+                        Symbol::new(&val, true),
                     );
                 }
 
@@ -565,7 +565,7 @@ impl Interpreter {
                     .last_mut()
                     .unwrap()
                     .symbols
-                    .insert(name.clone().value.into_string(), Symbol::new(class, false));
+                    .insert(name.clone().value.into_string(), Symbol::new(&class, false));
 
                 res.success(class)
             }
@@ -602,7 +602,7 @@ impl Interpreter {
         for idx in (0..self.ctx.len()).rev() {
             let sym = self.ctx.get(idx).unwrap().symbols.get(&k);
             if sym.is_some() {
-                return Some(sym.unwrap());
+                return Some((*sym.unwrap()).clone());
             }
         }
         None

--- a/src/core/interpreter/value.rs
+++ b/src/core/interpreter/value.rs
@@ -431,7 +431,7 @@ impl Value {
 
                 inter.ctx.last_mut().unwrap().symbols.insert(
                     name.clone().value.into_string(),
-                    Symbol::new(val.clone(), true),
+                    Symbol::new(&val, true),
                 );
             }
             let result = inter.interpret_node(body_node);
@@ -552,19 +552,6 @@ impl Value {
 
             let mut properties_e: HashMap<String, Value> = HashMap::new();
 
-            let soul = Self::Object {
-                pos_start,
-                pos_end,
-                properties: properties_e,
-            };
-
-            inter
-                .ctx
-                .last_mut()
-                .unwrap()
-                .symbols
-                .insert("soul".to_string(), Symbol::new(soul, false));
-
             for (k, v) in &properties {
                 let e_v = inter.interpret_node(v.clone());
                 if e_v.should_return() {
@@ -584,9 +571,22 @@ impl Value {
 
             properties_e.extend(methods_e);
 
+            let soul = Self::Object {
+                pos_start,
+                pos_end,
+                properties: properties_e,
+            };
+
+            inter
+                .ctx
+                .last_mut()
+                .unwrap()
+                .symbols
+                .insert("soul".to_string(), Symbol::new(&soul, false));
+
             let mut popped = false;
 
-            let soul = inter.ctx.last().unwrap().symbols.get("soul").unwrap().value;
+            let soul = inter.ctx.last().unwrap().symbols.get("soul").unwrap().value.clone();
 
             if constructor.is_some() {
                 let e_c = inter.interpret_node(constructor.unwrap());

--- a/src/std/lib.rs
+++ b/src/std/lib.rs
@@ -62,7 +62,7 @@ fn std_add_func(
     ctx.symbols.insert(
         name.to_string(),
         Symbol::new(
-            Value::InBuiltFunction {
+            &Value::InBuiltFunction {
                 name: name.to_string(),
                 pos_start: pos,
                 pos_end: pos,

--- a/src/utils/symbol.rs
+++ b/src/utils/symbol.rs
@@ -7,9 +7,9 @@ pub struct Symbol {
 }
 
 impl Symbol {
-    pub fn new(value: Value, reassignable: bool) -> Self {
+    pub fn new(value: &Value, reassignable: bool) -> Self {
         Self {
-            value,
+            value: (*value).clone(),
             reassignable,
         }
     }

